hw_33_online_shop/js/helpers.js

```diff
@@ -0,0 +1,227 @@
+export const Helpers = {
+
+    async getData(request) {
+
+        let data;
+
+        try {
+            let response = request;
+            data = await response.json();
+        
+            if (!response.ok) {
+                throw new Error(`${data.error}: ${data.message}`);
+        
+            }
+        } catch (error) {
+            console.log(error);
+        }
+
+        return data;
+    },
+
+    async getAllProducts() {
+
+        let products = getData(await fetch (`https://api.escuelajs.co/api/v1/products?offset=0&limit=10`));
+
+        return products;
+    },
+
+    async createNewProduct(newProduct) {
+
+        let updatedProduct = getData(
+            await fetch (`https://api.escuelajs.co/api/v1/products/`,
+                {
+                    method: "POST",
+                    headers: {
+                        "Content-type": "application/json"
+                    },
+                    body: JSON.stringify(newProduct)
+                }
+        ))
+
+        return updatedProduct;
+    },
+
+    async updateProduct(oldProduct, newProduct) {
+        
+        let updatedProduct = getData(
+            await fetch (`https://api.escuelajs.co/api/v1/products/${oldProduct.id}`,
```

Comment:
need to split the URL and save to variable (f.e. object) in order to not repeat yourself